tests/rules/invalid-field-name/invalid/component-models.json

```diff
@@ -0,0 +1,102 @@
+[
+  {
+    "id": "model-1",
+    "fields": [
+      {
+        "name": "my node",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "  leadingSpace",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "trailingSpace ",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "1node",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "my/node",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "my:node",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "my[node]",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "my|node",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "my*node",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "my\\backslash",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": ":quote",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": ".onedot",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "node\u0000name",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "node\tname",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "node\nname",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "🎉",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "_grouping",
+        "component": "text",
+        "label": "Text"
+      },
+      {
+        "name": "abc_group1_group2",
```

Comment:
we theoretically support this as we only treat the part before the first `_` as group name. everything after is just a property name. however, I think this rule makes still sense as it enforces not to mix concerns. like it.